### PR TITLE
For administrators, be able to create triggers and digests on user activity (#3162)

### PR DESCRIPTION
**Pull request by richard-julien**

### CODE REVIEW

**Comment by Archidoit** [comment]
What about adding recipients in activity triggers lines?
image

File: opencti-platform/opencti-front/src/private/components/common/form/ObjectMembersField.tsx [hunk 1]
```diff
@@ -89,13 +93,14 @@ const ObjectMembersField: FunctionComponent<ObjectMembersFieldProps> = ({
...
+        multiple={multiple ?? false}
         textfieldprops={{
           variant: 'standard',
-          label: t('Users, groups or organizations'),
+          label: t(label ?? 'Users, groups or organizations'),
           helperText: helpertext,
           onFocus: searchMembers,
         }}
-        onChange={onChange}
+        onChange={typeof onChange === 'function' ? onChange : null}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/common/form/ObjectMembersField.tsx:103]: () => onChange?.()

File: opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipEdition.jsx [hunk 2]
```diff
@@ -72,4 +61,12 @@ const StixCoreRelationshipEdition = (props) => {
   );
 };
 
+const StixCoreRelationshipEdition = ({ stixCoreRelationshipId, open, handleClose, handleDelete, noStoreUpdate }) => {
+  if (stixCoreRelationshipId && open) {
+    return <StixCoreRelationshipEditionInner open={open} stixCoreRelationshipId={stixCoreRelationshipId}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipEdition.jsx:66]: Can you align that better ? :D

File: opencti-platform/opencti-front/src/private/components/events/stix_sighting_relationships/StixSightingRelationshipEdition.jsx [hunk 3]
```diff
@@ -30,54 +30,39 @@ export const stixSightingRelationshipEditionDeleteMutation = graphql`
...
+            </React.Suspense>
+          )}
       </Drawer>
   );
 };
 
+// Workaround to prevent direct loading
+const StixSightingRelationshipEdition = ({ stixSightingRelationshipId, open, handleClose, handleDelete, inferred, noStoreUpdate }) => {
+  if (stixSightingRelationshipId && open) {
+    return <StixSightingRelationshipEditionInner open={open} stixSightingRelationshipId={stixSightingRelationshipId}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/events/stix_sighting_relationships/StixSightingRelationshipEdition.jsx:61]: Same

**Comment by Archidoit** [comment]
Error at edition of an activity trigger from the 'trigger and digest' tab 

Steps to reproduce:
Create an activity trigger named T1 with recipient = user1.
Log as user1.
Go in the trigger & digest tab. Update T1.
In the update form, click in the 'triggering on' field.
An error occurs.

File: opencti-platform/opencti-front/src/private/components/profile/triggers/TriggerLiveCreation.tsx [hunk 4]
```diff
@@ -254,14 +229,99 @@ const TriggerLiveCreation: FunctionComponent<TriggerLiveCreationProps> = ({
...
-  const liveFields = (
-    setFieldValue: (
-      field: string,
-      value: unknown,
-      shouldValidate?: boolean | undefined
-    ) => void,
-    values: TriggerLiveAddInput,
-  ) => (
+
+  const renderKnowledgeTrigger = (values: TriggerLiveAddInput, setFieldValue: (key: string, value: { value: string, label: string }[]) => void) => {
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/profile/triggers/TriggerLiveCreation.tsx:233]: value: Option

**Comment by Archidoit** [comment]
To do (known):
- update a trigger from the Activity trigger section
- translations: 'activity triggers', 'Create a live activity trigger', 'recipients', 'event scope', 'Create a regular activity digest',

**Pull request merged**